node/src/components/contract_runtime/error.rs

```diff
@@ -181,4 +182,6 @@ pub enum BlockExecutionError {
     InvalidTransactionArgs,
     #[error("Data Access Layer conflicts with chainspec setting: {0}")]
     InvalidAESetting(bool),
+    #[error("Catastrophic failure raised by VM2: {0}")]
+    Vm2InternalError(InternalHostError),
```

Comment:
it seems odd to me that we don't need / have a internal error for VM1 at this layer of the logic, but somehow do need it for VM2. For VM1 we just register the artifacts of an error result and keep processing. It's non-fatal.

I'd like to discuss it in standup to understand the reasoning here and discuss the bigger picture.